app/controllers/PdfDownloadController.scala

```diff
@@ -48,9 +48,9 @@ class PdfDownloadController @Inject() (
               url     <- OptionT.fromOption[Future](meta.url)
               content <- OptionT(fileStore.downloadFile(url))
             } yield Ok
-              .streamed(content, None, Some(meta.mimeType))
+              .streamed(content, None, meta.mimeType)
               .withHeaders(
-                "Content-Disposition" -> s"attachment; filename=${meta.fileName}"
+                "Content-Disposition" -> s"attachment; filename=${meta.fileName.getOrElse("New Attachment")}"
               )
```

Comment:
This determines the actual filename that will be downloaded - I think it might be better to do something like this:

```scala
           val pdfResult = for {
              pdf      <- OptionT.fromOption[Future](decision.decisionPdf)
              meta     <- OptionT(fileStore.getFileMetadata(pdf.id))
              url      <- OptionT.fromOption[Future](meta.url)
              fileName <- OptionT.fromOption[Future](meta.fileName)
              content  <- OptionT(fileStore.downloadFile(url))
            } yield Ok
              .streamed(content, None, Some(meta.mimeType))
              .streamed(content, None, meta.mimeType)
              .withHeaders(
                "Content-Disposition" -> s"attachment; filename=${fileName}"
              )
```
Note the additional generator in the for comprehension - this is because Upscan must have sent us a callback with the file info by the point that we are able to download a file